### PR TITLE
Add near text to GRPC and refactor nearText modules

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:

- Adds support for nearText
- Refactor nearTextParameters to be shared between different modules

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: grpc/weaviate.proto [hunk 1]
```diff
@@ -158,6 +159,19 @@ message HybridSearchParams {
...
 }
 
+message NearTextSearchParams{
+  message Move{
+    float force =1;
+    repeated string concepts = 2;
+    repeated string uuids = 3;
+  }
+
+  repeated string queries = 1;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:169]: what are queries? shouldn't be this `repeated string concepts`?

File: grpc/weaviate.proto [hunk 2]
```diff
@@ -158,6 +159,19 @@ message HybridSearchParams {
...
+  optional double certainty = 2;
+  optional double distance = 3;
+  Move move_to = 5;
+  Move move_away = 6;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:173]: I would also mark this as `optional`, and please fix the numbers :) should be 4,5 not 5,6

File: grpc/weaviate.proto [hunk 3]
```diff
@@ -158,6 +159,19 @@ message HybridSearchParams {
   FusionType fusion_type = 5;
 }
 
+message NearTextSearchParams{
+  message Move{
+    float force =1;
+    repeated string concepts = 2;
+    repeated string uuids = 3;
+  }
```
[the remark this answers is not in the document]
Comment (antas-marcin) [reply grpc/weaviate.proto:167]: you can make float also `double` as `certainty` and `distance` are

File: grpc/weaviate.proto [hunk 4]
```diff
@@ -158,6 +159,19 @@ message HybridSearchParams {
   FusionType fusion_type = 5;
 }
 
+message NearTextSearchParams{
```
Comment (antas-marcin) [inline grpc/weaviate.proto:162]: Maybe we could split this big proto file into smaller ones? I'm thinking for example about separate proto file for `filters` only `filters.proto`, we can do this later, but wanted to know your opinion on this

File: grpc/weaviate.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (dirkkul) [reply grpc/weaviate.proto:167]: force is not optional, and and a float value in the `ExploreMove` definition

File: grpc/weaviate.proto (hunk 4 above)
Comment (dirkkul) [reply grpc/weaviate.proto:162]: yes, good idea! Not in this PR, but later

File: grpc/weaviate.proto [hunk 5]
```diff
@@ -68,6 +68,7 @@ message SearchRequest {
   string after = 11;
   string tenant = 12;
   Filters filters = 13;
+  NearTextSearchParams near_text = 14;
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply grpc/weaviate.proto]: done, the other fields might need changes in the python client and might break the old client. I'd do this in a different PR

File: grpc/weaviate.proto (hunk 1 above)
Comment (dirkkul) [reply grpc/weaviate.proto]: After some discussion we decided to rename this from `concepts` to query. I always found that word unfitting - you're not looking for a concept, you're looking for something that is close to the text/query that you are providing.

For the Move* parameter, we decided to stay with concepts. While it also just a piece of text, it feels similar right to use here. You are searching for a *text*, but want to move away from this concept

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**